AMW_angular/io/src/app/settings/functions/function-edit.component.html

```diff
@@ -0,0 +1,30 @@
+<div class="modal-header">
+  <h5 class="modal-title" id="editFunctionTitle">{{ getTitle() }}</h5>
+  <button type="button" class="btn btn-light close" aria-label="Close" (click)="cancel()">
+    <span aria-hidden="true">&times;</span>
+  </button>
+</div>
+<div class="modal-body">
+  <div class="form-horizontal mx-2">
+    <div class="mb-3">
+      <label for="name" class="form-label">Function name</label>
+      <input type="text" class="form-control" id="name" [(ngModel)]="function.name" />
+    </div>
+    <div class="mb-3"> <label class="form-label">Function content</label>
+      <ngx-codemirror class="h-100"
+        [(ngModel)]="function.content"
+        [options]="{
+          lineNumbers: true,
+          theme: 'default',
+          mode: 'simplemode'
+         }"
+      ></ngx-codemirror>
+    </div>
+  </div>
+</div>
+<div class="modal-footer">
+  <button type="button" class="btn btn-light" (click)="cancel()">Cancel</button>
+  @if (canManage && function.name.length > 0) {
```

Comment:
- Improve usability by providing a tooltip which informs the user what is missing. 
- Make button disabled instead of hidden. -> only hide the button, if he is not privileged
- Validate both parts of a function not only the name (-> according the old UI)